generate_sample_file.py

```diff
@@ -135,36 +135,74 @@ def generate_sample_file(self, output_file_path: Path, treatment_code_quantities
-                        'UPRN': self.get_random_uprn(),
-                        'ADDRESS_TYPE': 'CE' if community_establishment else "HH",
-                        'ESTAB_TYPE': self.get_random_ce_type() if community_establishment else 'Household',
-                        'ADDRESS_LEVEL': 'E' if community_establishment else 'U',
-                        'ABP_CODE': self.get_random_abp_code(),
-                        'ORGANISATION_NAME': '',
-                        'ADDRESS_LINE1': self.get_random_address_line(),
-                        'ADDRESS_LINE2': '',
-                        'ADDRESS_LINE3': '',
-                        'TOWN_NAME': self.get_random_post_town(),
-                        'POSTCODE': self.get_random_post_code(),
-                        'LATITUDE': self.get_random_lat_or_long(),
-                        'LONGITUDE': self.get_random_lat_or_long(),
-                        'OA': self.get_random_regiony_type_thing(),
-                        'LSOA': self.get_random_regiony_type_thing(),
-                        'MSOA': self.get_random_regiony_type_thing(),
-                        'LAD': self.get_random_regiony_type_thing(),
-                        'REGION': self.get_random_regiony_type_thing(),
-                        'HTC_WILLINGNESS': self.get_random_htc(),
-                        'HTC_DIGITAL': self.get_random_htc(),
-                        'TREATMENT_CODE': item["treatment_code"],
-                        'FIELDCOORDINATOR_ID': '',
-                        'FIELDOFFICER_ID': '',
-                        'CE_EXPECTED_CAPACITY': self.get_random_ce_capacity() if community_establishment else '',
-                    })
+
+                    # randomly decide whether to create a CE case or not
+                    if self.random_1_in_11():
+                        self._write_estab_case(writer, sequential_arid, item)
+                        continue
+
+                    # otherwise write a non CE case
+                    self._write_row(writer, sequential_arid, False, 'U', item,
+                                    0)  # have to change this when introducing SPG cases as they can be 'E' or 'U'
+
+    def _write_estab_case(self, writer, sequential_arid, item):
+        #  randomly decide if we want to create a CE/U case
+        if self.random_1_in_11():
+            self._write_parent_and_unit_cases(writer, sequential_arid, item)
+            return
+
+        # otherwise write CE/E case
+        self._write_row(writer, sequential_arid, True, 'E', item, self.get_random_ce_capacity())
+
+    def _write_parent_and_unit_cases(self, writer, sequential_arid, item):
+        # create parent case
+        parent_arid, estab_type = self._write_row(writer, sequential_arid, True, 'E', item, 0)
+
+        # create child cases
+        for _ in range(10):
+            self._write_row(writer, sequential_arid, True, 'U', item, self.get_random_ce_capacity(), parent_arid,
```

Comment:
What is item? 
I think this might be better to use kwargs after `sequential_arid`. Don't know what they're all used for so having `Community_level=U''`and so on might be better